src/test/java/bio/terra/endpoints/CLICommandTests.java

```diff
@@ -0,0 +1,284 @@
+package bio.terra.endpoints;
+
+import bio.terra.common.category.CLIIntegrated;
+import bio.terra.context.Login;
+import org.junit.*;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import java.util.*;
+
+@RunWith(JUnit4.class)
+@Category(CLIIntegrated.class)
+public class CLICommandTests {
+
+    @BeforeClass
+    public static void setup() {
+        Login.setClientSecretsFilePath(CLITestingConfig.clientSecretsFilePath);
+    }
+
+    @AfterClass
+    public static void teardown() {
+        Login.setClientSecretsFilePath(null);
+    }
+
+    /**
+     * Test method that calls commands in sequence so data gets setup in the right order.
+     * @throws IOException
+     */
+    @Test
+    public void commandScript() throws IOException {
+        // fetch access token in the same way that the CLI does
+        // this depends on the jadecli_client_secret.json file
+        Login.authorize();
+        String token = Login.getUserCredential().getAccessToken();
+
+        // TODO: create and delete profile, need to modify input JSON file for create dataset and write to /tmp
+
+        // jc dataset create --input-json inputDatasetCreate.txt
+        Map<String, Object> datasetDetails =
+                datasetCreateTest(token, "inputDatasetCreate.txt", "outputDatasetCreate.txt");
+
+        // jc dataset show CLITestDataset
+        datasetShowTest(datasetDetails, "outputDatasetShow.txt");
+
+        // jc dr describe CLITestDataset
+        drDescribeTest(datasetDetails, "outputDatasetShow.txt");
+
+        // jc dataset delete CLITestDataset
+        datasetDeleteTest(datasetDetails, "outputDatasetDelete.txt");
+    }
+
+    /**
+     * CLI command : jc dataset create
+     * Repository API : POST : createDataset
+     * @throws IOException
+     */
+    public Map<String, Object> datasetCreateTest(String token, String inputJSONFilename, String expectedOutputFilename)
+            throws IOException {
+        System.out.println("***********************************************");
+        System.out.println("jc dataset create --input-json inputDatasetCreate.txt");
+
+        // build path to input file
+        String inputJSON = CLITestingConfig.dirName + inputJSONFilename;
+
+        // call CLI command in a separate process
+        List<String> cmdArgs = new ArrayList<String>(new ArrayList<>(Arrays.asList(
+                new String[]{"dataset", "create", "--input-json", inputJSON})));
+        List<String> cliCmdResponse = CLITestingUtils.callCLICommand(cmdArgs);
+
+        // read in the expect CLI command output from a file
+        List<String> cliCmdExpectedResponse = CLITestingUtils.readCLIExpectedOutput(expectedOutputFilename);
+
+        // log the response to stdout
```

Comment:
Maybe use logger here so we can control how chatty the test is?

Reply:
Good idea. Done.